client/otagent/src/main/java/org/hawkular/apm/otagent/propagation/MapExtractAdapter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.otagent.propagation;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import io.opentracing.propagation.TextMap;
+
+/**
+ * @author gbrown
+ */
+public final class MapExtractAdapter implements TextMap {
```

Comment:
They are default implementations that could be used with BTM rules that propagate headers using Map. Not used by the current rules, but are there for convenience.